File: lib/detectors/authTokens.js
module.exports = {
  name: 'Authorization Tokens',
  severity: 'high',
  run(content) {
    const out = [];
    const bearerRe = /Bearer\s+([A-Za-z0-9\-_.=\/]+)/g;
    const authRe = /authorization\s*["']?\s*[:=]\s*["']([^"']{8,200})["']/i;
    let m;
    const seen = new Set();
    while ((m = bearerRe.exec(content)) !== null) {
      const val = m[0];
      if (seen.has(val)) continue;
      seen.add(val);
      const idx = m.index;
      const context = content.substr(Math.max(0, idx - 40), 120).replace(/\n/g, ' ');
      out.push({ value: val, context, severity: 'high' });
    }

    const m2 = authRe.exec(content);
    if (m2 && !seen.has(m2[0])) {
      out.push({ value: m2[1], context: content.substr(Math.max(0, m2.index - 40), 120).replace(/\n/g, ' '), severity: 'high' });
    }

    return out;
  }
};

File: JSX/detectors/authTokens.py
"""Authorization token detector."""

import re
from .base import BaseDetector


class Detector(BaseDetector):
    """Detects authorization tokens in JavaScript."""

    def __init__(self):
        super().__init__("Authorization Tokens", severity="high")
        self.bearer_pattern = re.compile(r'Bearer\s+([A-Za-z0-9\-_.=/]+)')
        self.auth_pattern = re.compile(r'authorization\s*["\']?\s*[:=]\s*["\']([^"\']{8,200})["\']', re.IGNORECASE)

    def run(self, content):
        """Find all authorization tokens."""
        findings = []
        seen = set()

        # Look for Bearer tokens
        for match in self.bearer_pattern.finditer(content):
            # capture the token part (group 1)
            value = match.group(1)
            if value in seen:
                continue
            seen.add(value)

            context = self._get_context(content, match.start())
            findings.append({
                "value": value,
                "context": context,
                "pos": match.start(),
                "severity": "high"
            })

        # Look for authorization header assignments
        for match in self.auth_pattern.finditer(content):
            value = match.group(1)
            if value in seen:
                continue
            seen.add(value)

            context = self._get_context(content, match.start())
            findings.append({
                "value": value,
                "context": context,
                "pos": match.start(),
                "severity": "high"
            })

        return findings
